src/app-layer-ssl.c

```diff
@@ -1171,6 +1192,98 @@ static inline int TLSDecodeHSHelloExtensionEllipticCurvePF(SSLState *ssl_state,
     return -1;
 }
 
+static inline int TLSDecodeHSHelloExtensionSigAlgorithms(
+        SSLState *ssl_state, const uint8_t *const initial_input, const uint32_t input_len)
+{
+    const uint8_t *input = initial_input;
+
+    /* Empty extension */
+    if (input_len == 0)
+        return 0;
+
+    if (!(HAS_SPACE(2)))
+        goto invalid_length;
+
+    uint16_t sigalgo_len = (uint16_t)(*input << 8) | *(input + 1);
+    input += 2;
+
+    if (!(HAS_SPACE(sigalgo_len)))
+        goto invalid_length;
+
+    if ((ssl_state->current_flags & SSL_AL_FLAG_STATE_CLIENT_HELLO) &&
+            SC_ATOMIC_GET(ssl_config.enable_ja4)) {
+        uint16_t sigalgo_processed_len = 0;
+        /* coverity[tainted_data] */
+        while (sigalgo_processed_len < sigalgo_len) {
```

Comment:
infinite loop if sigalgo_len = 0xFFFF
We overflow uint16 going from 0xFFFE to 0 and looping forever

Reply:
Right! We need to check if `sigalgo_len % 2 == 0` and fail otherwise without trying to parse further. This is also done in other similar places in the C parsers, preventing such bugs. Will add this in the code.

Reply:
Done